.github/workflows/ci.yml

```diff
@@ -728,20 +727,26 @@ jobs:
         python -m pip install --upgrade pip
         python -m pip install pytest
         python -m pip install pytest-xdist
+        python -m pip install numba
         echo ::endgroup::
+
+        echo ::group::Run complete test suite
+        python -m pytest || true
```

Comment:
Ok, I think I understand now. This command runs also the xfailed. If an xfailed test crashes then we abort. That's what happens with the osx builds. To make them work we should mark the crashing xfailed as crashing.

The other reason for the remaining failures is the valgrind complaints on the tests where we call a void function. We should fix that...

Reply:
Yes but currently `python -m pytest` skips the crashing xfails since we have marked them as `run = False` so it is already taken care of. The current failures are in the groups - xfail logs and passing test logs (with valgrind) since they seem to bring up the errors.

I will update the CI to handle the same and we should be able to drop `|| true` when we run the whole test suite with `python -m pytest`